source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -156,6 +158,51 @@ void TextureBaker::bakeGraphOutput(OutputPtr output, GenContext& context, const
     _bakedImageMap[output].push_back(baked);
 }
 
+
+void TextureBaker::renderTextureSpace()
+{
+    GlslProgramPtr program = getProgram();
+    GLFrameBufferPtr framebuffer = getFrameBuffer();
+    unsigned int bakedTextureHeight = _height;
+    unsigned int bakedTextureWidth = _width;
+
+    if (_autoTextureResolution)
+	{
+		// Prefetch all required images and query thier dimensions. 
+		// Since Images are cached by ImageHandler, they will be reused during bindTextures
+		const GlslProgram::InputMap& uniformList = program->getUniformsList();
+		for (const auto& uniform : uniformList)
+		{
+			GLenum uniformType = uniform.second->gltype;
+			GLint uniformLocation = uniform.second->location;
+			if (uniformLocation >= 0 && uniformType >= GL_SAMPLER_1D && uniformType <= GL_SAMPLER_CUBE)
+			{
+				const string fileName(uniform.second->value ? uniform.second->value->getValueString() : "");
+				if (fileName != HW::ENV_RADIANCE &&
+					fileName != HW::ENV_IRRADIANCE)
+				{
+					ImagePtr image = _imageHandler->acquireImage(fileName);
+					if (image)
+					{
+						const unsigned int imageHeight = image->getHeight();
+						const unsigned int imageWidth = image->getWidth();
+						bakedTextureHeight = imageHeight > bakedTextureHeight ? imageHeight : bakedTextureHeight;
+						bakedTextureWidth = imageWidth > bakedTextureWidth ? imageWidth : bakedTextureWidth;
+					}
+				}
+			}
+		}
+	}
+
+    program->bind();
```

Comment:
I think you can perform the resize first and the call (parent class) GlslRenderer::renderTextureSpace() to avoid duplicate code.
Also avoid accidently having code become out of sync or missing some step :).

Reply:
Alternative is to add a "compute framebuffer w/h" override method and then call GlslRenderer::renderTextureSpace().